File: resources/[soz]/soz-weather/time.lua
currentHour = 12
currentMinute = 00
currentSecond = 00

-- How many seconds in real time to do 24h in GTA
dayInSeconds = 1800
clockTick = 1000
incrementSeconds = ((3600 * 24) / dayInSeconds) / (1000 / clockTick)

function AdvanceTime()
    currentSecond = currentSecond + incrementSeconds

    if currentSecond >= 60 then
        incrementMinutes = math.floor((currentSecond / 60) + 0.5)
        currentMinute = currentMinute + incrementMinutes
        currentSecond = currentSecond % 60

        if currentMinute >= 60 then
            incrementHours = math.floor((currentMinute / 60) + 0.5)
            currentMinute = currentMinute % 60
            currentHour = currentHour + incrementHours

            if currentHour >= 24 then
                currentHour = currentHour % 24
            end
        end
    end
end

File: resources/[soz]/soz-phone/src/nui/apps/weather/utils/time.ts
type Time = {
    hour: number;
    minute: number;
    second: number;
};

export const extractTime = (time: string) => {
    const [hour, minute] = time.split(':');
    return { hour: Number(hour), minute: Number(minute), second: 0 };
};

export const advanceTime = (time: Time, durationInMs: number) => {
    const incrementSeconds = durationInMs / 1000;
    time.second += incrementSeconds;
    if (time.second >= 60) {
        const incrementMinutes = Math.floor(time.second / 60);

        time.minute += incrementMinutes;
        time.second %= 60;

        if (time.minute >= 60) {
            const incrementHours = Math.floor(time.minute / 60);

            time.hour += incrementHours;
            time.minute %= 60;

            if (time.hour >= 24) {
                time.hour %= 24;
            }
        }
    }
    return time;
};

export const isDay = (time: Time) => {
    return time.hour >= 6 && time.hour < 21;
};
